src/commands/plugins.rs

```diff
@@ -0,0 +1,305 @@
+        if uninstalled {
+            println!("Plugin {} was successfully uninstalled", self.name);
+        } else {
+            println!(
+                "Plugin {} isn't present, so no changes were made",
+                self.name
+            );
+        }
+        Ok(())
+    }
+}
+
+#[derive(Parser, Debug)]
+pub struct Upgrade {
+    /// Name of Spin plugin to upgrade.
+    #[clap(
+        name = PLUGIN_NAME_OPT,
+        conflicts_with = PLUGIN_ALL_OPT,
+        required_unless_present_any = [PLUGIN_ALL_OPT],
+    )]
+    pub name: Option<String>,
+
+    /// Upgrade all plugins.
+    #[clap(
+        short = 'a',
+        long = "all",
+        name = PLUGIN_ALL_OPT,
+        conflicts_with = PLUGIN_NAME_OPT,
+        conflicts_with = PLUGIN_REMOTE_PLUGIN_MANIFEST_OPT,
+        conflicts_with = PLUGIN_LOCAL_PLUGIN_MANIFEST_OPT,
+        takes_value = false,
+    )]
+    pub all: bool,
+
+    /// Path to local plugin manifest.
+    #[clap(
+        name = PLUGIN_LOCAL_PLUGIN_MANIFEST_OPT,
+        short = 'f',
+        long = "file",
+        conflicts_with = PLUGIN_REMOTE_PLUGIN_MANIFEST_OPT,
+    )]
+    pub local_manifest_src: Option<PathBuf>,
+
+    /// Path to remote plugin manifest.
+    #[clap(
+        name = PLUGIN_REMOTE_PLUGIN_MANIFEST_OPT,
+        short = 'u',
+        long = "url",
+        conflicts_with = PLUGIN_LOCAL_PLUGIN_MANIFEST_OPT,
+    )]
+    pub remote_manifest_src: Option<Url>,
+
+    /// Skips prompt to accept the installation of the plugin[s].
+    #[clap(short = 'y', long = "yes", takes_value = false)]
+    pub yes_to_all: bool,
+
+    /// Specific version of a plugin to be install from the centralized plugins
+    /// repository.
+    #[clap(
+        long = "version",
+        short = 'v',
+        conflicts_with = PLUGIN_REMOTE_PLUGIN_MANIFEST_OPT,
+        conflicts_with = PLUGIN_LOCAL_PLUGIN_MANIFEST_OPT,
+        conflicts_with = PLUGIN_ALL_OPT,
+        requires(PLUGIN_NAME_OPT)
+    )]
+    pub version: Option<Version>,
+
+    /// Allow downgrading a plugin's version.
+    #[clap(short = 'd', long = "downgrade", takes_value = false)]
+    pub downgrade: bool,
+}
+
+impl Upgrade {
+    /// Upgrades one or all plugins by reinstalling the latest or a specified
+    /// version of a plugin. If downgrade is specified, first uninstalls the
+    /// plugin.
+    pub async fn run(self) -> Result<()> {
+        let manager = PluginManager::default()?;
+        let manifests_dir = manager.store().installed_manifests_directory();
+
+        // Check if no plugins are currently installed
+        if !manifests_dir.exists() {
+            println!("No currently installed plugins to upgrade.");
+            return Ok(());
+        }
+
+        if self.all {
+            self.upgrade_all(manifests_dir).await
+        } else {
+            let plugin_name = self
+                .name
+                .clone()
+                .ok_or_else(|| anyhow!("plugin name is required for upgrades"))?;
```

Comment:
As a nice convenience, `anyhow::Context` is also implemented for `Option`.
```suggestion
                .context("plugin name is required for upgrades")?;
```